linbox/matrix/densematrix/blas-matrix.h

```diff
@@ -334,155 +166,107 @@ namespace LinBox
 		Self_t& operator= (const Self_t& A) ;
 
-		/// Make this a (deep)copy of B. Assumes same shape.
-		//! make sure we actually copy
-		template<class Matrix>
-		BlasMatrix &copy( const Matrix & B)
-		{
-			Element x; field().init(x);
-			for (size_t i = 0 ; i < rowdim() ; ++i)
-				for (size_t j = 0 ; j < coldim() ; ++j) {
-					setEntry(i,j,B.getEntry(x,i,j));
-				}
-			return *this;
-
-		}
-
+        //! operator = (copying data from different matrix type)
+        template<class _Matrix>
+        Self_t& operator= (const _Matrix& A) ;
 
+        
 		//! Rebind operator
 		template<typename _Tp1>
 		struct rebind ;
 
-		//////////////////
-		//  DIMENSIONS  //
-		//////////////////
+		/////////////////
+		//  ACCESSORS  //
+		/////////////////
 
 		/** Get the number of rows in the matrix.
 		 * @returns Number of rows in matrix
 		 */
-		size_t rowdim() const ;
+		size_t rowdim() const { return _row;}
 
 		/** Get the number of columns in the matrix.
 		 * @returns Number of columns in matrix
 		 */
-		size_t coldim() const ;
+		size_t coldim() const { return _col;}
 
-		/*! Get the stride of the matrix.
-		 */
-		size_t getStride() const;
-		size_t stride() const { return getStride() ;}
-
-		/*!Get a reference to the stride of the matrix.
-		 * Modify stride this way.
-		 */
-		size_t& getWriteStride();
+		/*! Get the stride of the matrix.*/		 
+		size_t getStride() const { return _col;}
 
-
-		/** Resize the matrix to the given dimensions.
-		 * The state of the matrix's entries after a call to this method is
-		 * undefined
-		 * @param m Number of rows
-		 * @param n Number of columns
-		 * @param val
+		/*! @internal
+		 * Get read-only access to the matrix data.
 		 */
-		void resize (const size_t &m, const size_t &n, const Element& val = Element()) ;
-
+	    Element_ptr getPointer() const { return _ptr;}
+        Element_ptr& getWritePointer()  { return _ptr;}
+        
 		//////////////////
 		//   ELEMENTS   //
 		//////////////////
 
-		/*! @internal
-		 * Get read-only pointer to the matrix data.
-		 */
-		pointer getPointer() const ;
-
-		const_pointer &getConstPointer() const ;
-
-		Rep & refRep() { return _rep ;}
-		const Rep & getRep() const { return _rep ;}
-
-
-		/*! @internal
-		 * Get write pointer to the matrix data.
-		 * Data may be changed this way.
-		 */
-		pointer& getWritePointer() ;
 
 		/** Set the entry at the (i, j) position to a_ij.
 		 * @param i Row number, 0...rowdim () - 1
 		 * @param j Column number 0...coldim () - 1
 		 * @param a_ij Element to set
 		 */
-		const Element& setEntry (size_t i, size_t j, const Element &a_ij) ;
+		const Element& setEntry (size_t i, size_t j, const Element &a_ij)
+        {
+            field().assign(_rep[i*_col+j],a_ij); return _rep[i*_col+j]; // PG: this does not work with BitVector as _rep -> getting a Warning because BitVector operator[] does not return a reference but a boolean constructed on the fly (setEntry must return a reference [why ?])
```

Comment:
That's indeed a design problem. `setEntry` should probably not return a reference. We should open an issue.

Reply:
We discussed this with Pascal a while ago. I think setEntry should return `void`, but this means a rather strong change in the API of LinBox matrix. I vote for it.

Reply:
I also vote for it. I don't remember any reasons why our API for methods must similar to the operator's one.

Reply:
All right, opened #135. Discussions go there.